service/frontend/service.go

```diff
@@ -256,9 +251,7 @@ func (s *Service) Start() {
 	clusterMetadata := s.GetClusterMetadata()
 	if clusterMetadata.IsGlobalNamespaceEnabled() {
 		consumerConfig := clusterMetadata.GetReplicationConsumerConfig()
-		if consumerConfig != nil &&
-			consumerConfig.Type == config.ReplicationConsumerTypeRPC &&
-			s.config.EnableRPCReplication() {
+		if consumerConfig.Type == config.ReplicationConsumerTypeRPC || consumerConfig.Type == config.ReplicationConsumerTypeKafkaToRPC {
```

Comment:
I see this condition is repeated in many places.  Might be useful to create a helper method.

Reply:
this check will only exists for one version, i.e. the version which RPC stack is ready, then kafka will be removed in the next version and all `config.ReplicationConsumerType*` will be removed